packages/docusaurus-theme-classic/src/routeAnnouncer.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {translate} from '@docusaurus/Translate';
+import type {ClientModule} from '@docusaurus/types';
+import './routeAnnouncer.css';
+
+const clientModule: ClientModule = {
+  onRouteDidUpdate({location, previousLocation}) {
+    if (!previousLocation || location.pathname === previousLocation.pathname) {
+      return;
+    }
+    Array.from(
+      document.getElementsByTagName('docusaurus-route-announcer'),
+    ).forEach(document.removeChild);
```

Comment:
Clear the announcement from previous route transition. Technically, we can put it in the cleanup function of `onRouteDidUpdate`, but we don't have a clear semantic of what that cleanup should do.